### PR TITLE
feat(tracing): replace Jaeger with Tempo & integrate it with Grafana

### LINKED ISSUE #10236: Discussion: replace minstant and minitrace with the standard instant and tracing library
minitrace relies on minstant.

We have seen two issues related to the portability of `minstant` in different platforms: #4547 and #10235. 

IMO, the bottleneck won't be in the tracing and timing library in a foreseeable future. Fixing ande debugging portability issues in minstant and potentially minitrace can be a headache. I suggest we consider using more widely used libraries like tokio tracing and std time instead.

Discussions are welcomed.

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support Grafana Tempo in RiseDev as the replacement for Jaeger, since it has better integration with Grafana or Loki and is a candidate for tracing service in the cloud (https://github.com/risingwavelabs/risingwave/pull/10315#issuecomment-1596558976). Now developers can simply find the traces on the Grafana Dashboard at `localhost:3001`!

Currently, the dashboard is maintained as JSON and we may reuse the dashboard generator in the future if more panels are going to be added. 

Close #10236.

### Previews

Screenshot 2023-06-19 at 9 59 12 PM
Screenshot 2023-06-19 at 9 59 34 PM

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/tracing/src/lib.rs
Comment (xxchan) [inline src/tracing/src/lib.rs:1]: A little question: why this was needed before, but not needed now?

**Review by xxchan — COMMENTED** [review]
code changes generally LGTM

File: src/tracing/src/lib.rs
Comment (BugenZhao) [reply src/tracing/src/lib.rs:1]: With `minitrace-jaeger`, we have to manually collect spans and encode them and send them to jaeger, which is what this code snippet is for. `opentelemetry-otlp` helps us to do this.

https://github.com/risingwavelabs/risingwave/blob/4a9ec1ea17c410457e6b2e010620167765796b01/src/utils/runtime/src/lib.rs#L348-L367

**Review by fuyufjh — APPROVED** [review]
LGTM! cc. @Nebulazhang

**Review by arkbriar — APPROVED** [review]
LGTM!

**Comment by arkbriar** [comment]
To make it possible to distinguish between different RisingWave clusters, is there a way to add additional labels besides the service name for the trace records?

**Comment by BugenZhao** [comment]
> That means we'll need to add some unique label (or whatever it is called in tracing) to identify the RisingWave

Should the kernel directly attach these properties when reporting to the collector? IIUC currently we collect logs with middleware and cluster (or k8s) related properties are attached there.

**Comment by arkbriar** [comment]
> > That means we'll need to add some unique label (or whatever it is called in tracing) to identify the RisingWave
> 
> Should the kernel directly attach these properties when reporting to the collector? IIUC currently we collect logs with middleware and cluster (or k8s) related properties are attached there and I'm not sure whether we can adopt a similar approach.

That would be better. cc. @Nebulazhang

**Pull request merged**